internal/notification/handlers/notification_worker.go

```diff
@@ -79,24 +73,18 @@ func (w *NotificationWorker) Work(ctx context.Context, job *river.Job[*notificat
 	if err == nil {
 		return nil
 	}
-	// if retries are disabled or if the error explicitly specifies, we cancel the notification
-	// TODO: move max attempts to job config
-	if w.config.MaxAttempts <= 1 || errors.Is(err, &channels.CancelError{}) {
-		return river.JobCancel(errors.New("notification is too old"))
+	// if the error explicitly specifies, we cancel the notification
+	if errors.Is(err, &channels.CancelError{}) {
+		return river.JobCancel(err)
 	}
 	return err
 }
 
 type WorkerConfig struct {
 	LegacyEnabled       bool
 	Workers             uint8
-	RequeueEvery        time.Duration
-	RetryRequeueEvery   time.Duration
 	TransactionDuration time.Duration
-	MaxAttempts         uint8
 	MaxTtl              time.Duration
-	MinRetryDelay       time.Duration
-	MaxRetryDelay       time.Duration
 	RetryDelayFactor    float32
```

Comment:
I think we should definitely keep `MaxAttempts`. The issue is that we're [now using the legacy MaxFailureCount](https://github.com/zitadel/zitadel/blob/notification-queue/internal/notification/handlers/user_notifier.go#L114) instead, which I think is just a mistake. It might require additional refactoring, but keeping the legacy config out of the queue worker is worth it imo